app/src/test/java/waiter/ClientConnectionTest.java

```diff
@@ -0,0 +1,35 @@
+package waiter;
+
+import waiter.ClientConnection.ClientConnection;
+
+import org.junit.jupiter.api.Test;
+import waiter.InputStreamer.mock.InputStreamerMock;
+import waiter.OutputStreamer.mock.OutputStreamerMock;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+
+public class ClientConnectionTest {
+
+    @Test
+    void readsMessage() throws IOException {
+        String input = "foo";
+        InputStreamerMock inputStreamerMock = new InputStreamerMock(input);
+        OutputStreamerMock outputStreamerMock = new OutputStreamerMock();
+        ClientConnection clientConnection = new ClientConnection(inputStreamerMock, outputStreamerMock);
+        Message readMessage = clientConnection.read();
+        assertEquals(readMessage.open(), input);
+    }
+
+    @Test
+    void writesMessage() throws IOException {
+        String input = "foo";
+        Message outputMessage = new Message(input);
+        InputStreamerMock inputStreamerMock = new InputStreamerMock(input);
+        OutputStreamerMock outputStreamerMock = new OutputStreamerMock();
+        ClientConnection clientConnection = new ClientConnection(inputStreamerMock, outputStreamerMock);
```

Comment:
Similarly, consider inlining here.

Reply:
That makes sense, ignore the second part of my comment above 😂

Reply:
how about this:
```java
String input = "foo";
OutputStreamerMock outputStreamerMock = new OutputStreamerMock();

ClientConnection clientConnection = new ClientConnection(
  InputStreamerMock(foo),
  outputStreamerMock
);

assertEquals(outputStreamerMock.writtenOutput, input);
```

I feel like you don't really line 28, I could be wrong though.